Add rearranged version of RNA-seq clustering example notebook

cbethell opened the pull request:
## Purpose:
What issue(s) does your PR address?

This PR addresses the issue creating a clustering example with RNA-seq data (#127).

As noted in my [last comment on that issue](https://github.com/AlexsLemonade/refinebio-examples/issues/127#issuecomment-669347715), I followed a plan to reorder/shorten the clustering notebook in closed PR #140:

1. Choose a simpler/less complex dataset (removing the need for the collapse replicates steps).
2. Start by plotting the heat map.
3. Prepare metadata and add annotation bars.
4. Update sentence structure/context after draft PR is filed.

## Questions for reviewers

- Does the dataset seem like to be better suited for use in this example?
- Is the execution of steps still logical? 
- Should I be copying the introduction from the [microarray clustering notebook](https://github.com/AlexsLemonade/refinebio-examples/pull/122/files) into the introduction for this PR?

Note that I removed the numbering strategy and followed the other measures outlined with the contributing guidelines (_I believe_). However, the Snakemake function `snakemake --cores 1` won’t run when I add the notebook to the `Snakefile`, so I may be adding it wrong. I got it to work for the notebook in this PR by replacing what was already there and I didn’t want to commit and overwrite what’s in the repo.

## Pull Request Check List:
* [x] Run a linter
* [x] Set the seed (if applicable)
* [x] Comments and/or documentation up to date
* [x] Double check your paths
* [x] Check headline numbering
* [x] Spell check any Rmd file or md file
* [x] Restart R and run all notebooks fresh and save

cbethell commented:
@jaclyn-taroni and @cansavvy, I am requesting your review here to get your thoughts on the code and dataset that I implemented in the clustering notebook in this PR. 

I _believe_ that the notebook that I chose is the simpler of the [two that I had posted](https://github.com/AlexsLemonade/refinebio-examples/pull/140#issuecomment-668846394) but am worried there may be a need for some additional metadata cleaning here, specifically around the `refinebio_treatment` variable. I can get started on updating comments and context once we decide whether or not we do want to keep this dataset.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
````diff
@@ -0,0 +1,334 @@
+```
+
+### Perform DESeq2 normalization
+We are going to use the `rlog` function from the `DESeq2` package to normalize
````
```suggestion
We are going to use the `rlog()` function from the `DESeq2` package to normalize
```

This reminds me. I need to add this to the style guide. I think we want to stick with the notation @jashapiro came up with over on `training-modules` which is when we are talking about functions, always include the `()`. I think this helps users low-key keep track of what things are functions.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,334 @@
+  pheatmap(
+    df_by_var,
+    cluster_rows = TRUE,
+    cluster_cols = TRUE,
```
We need to add a `show_rownames = FALSE` on both heatmaps. The rownames are a bit smooshy and not really necessary.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,334 @@
+dds <- DESeqDataSetFromMatrix(
+  countData = df, # This is the data.frame with the counts values for all replicates in our dataset
+  colData = metadata, # This is the data.frame with the annotation data for the replicates in the counts data.frame
+  design = ~refinebio_treatment # Replace with an appropriate design variable for your analysis
```
I think we should probably include the `cancer_type` info in the model too (I haven't read too much about the set up, but it looks like the mutations and the drug treatments are two separate things that are being tested).

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,334 @@
+annotation_df <- metadata %>%
+  # Create a variable to store the cancer type information
+  dplyr::mutate(
+    cancer_type = dplyr::case_when(
```
Since the mice don't have cancer per se, it might be a little clearer to name this `mutation` or something like that. (Again, haven't read the paper so you may want to double check this).

cansavvy replied:
Second. thought is if we are including both the `cancer_type` and `refinebio_treatment` as labels in the heatmap, the we should include them both in the model in the `DESeq2` object. (Even though we aren't using it for DE here but its just good to set a good set up ).

cbethell replied:
> Second. thought is if we are including both the cancer_type and refinebio_treatment as labels in the heatmap, the we should include them both in the model in the DESeq2 object. (Even though we aren't using it for DE here but its just good to set a good set up ).

@cansavvy I get the following error when trying to implement this change: 
```
Error in checkFullRank(modelMatrix) : 
  the model matrix is not full rank, so the model cannot be fit as specified.
  One or more variables or interaction terms in the design formula are linear
  combinations of the others and must be removed.
```

Furthermore, before opening this revamped PR, @jaclyn-taroni suggested starting with the heatmap then preparing the annotation data.frame, noting that the metadata does not necessarily need to be cleaned up metadata for the `DESeqDataSet` object.

cansavvy replied:
Hmm.. Okay. Can you do a `table()` of those two variables? Perhaps there's not enough samples to use both variables, in which case, that's fine, we can just explain why we are choosing one over the other. 

I am a bit confused by this comment. Can you give me more context and/or a link?: 
> noting that the metadata does not necessarily need to be cleaned up metadata for the DESeqDataSet object.

cansavvy replied:
Also can you paste here the code or permalink to it that gives you that error mentioned above?

cbethell replied:
> Hmm.. Okay. Can you do a table() of those two variables? Perhaps there's not enough samples to use both variables, in which case, that's fine, we can just explain why we are choosing one over the other.

Would you like me to do this a commit the changes or would you prefer that I post the results of `table()` under this comment for discussion purposes @cansavvy?

> I am a bit confused by this comment. Can you give me more context and/or a link?:

This was discussed through direct messaging on Slack, here's a snippet of the convo:
<img width="595" alt="Screen Shot 2020-08-11 at 2 24 00 PM" src="https://user-images.githubusercontent.com/43576623/89934423-88a5a080-dbde-11ea-874e-00466bfd48a5.png">

cbethell replied:
> Also can you paste here the code or permalink to it that gives you that error mentioned above?

Adapted from lines 212-217:

```
# Create a `DESeqDataSet` object
dds <- DESeqDataSetFromMatrix(
  countData = df, # This is the data.frame with the counts values for all replicates in our dataset
  colData = annotation_df, # This is the data.frame with the annotation data for the replicates in the counts data.frame
  design = ~refinebio_treatment + mutation # Replace with an appropriate design variable for your analysis
)
```

I did not commit this change because of the error.

cbethell replied:
> Hmm.. Okay. Can you do a table() of those two variables? Perhaps there's not enough samples to use both variables, in which case, that's fine, we can just explain why we are choosing one over the other.

Here you go!
`table(annotation_df$refinebio_treatment, annotation_df$mutation)`
<img width="187" alt="Screen Shot 2020-08-11 at 2 33 22 PM" src="https://user-images.githubusercontent.com/43576623/89935137-ad4e4800-dbdf-11ea-9de6-dbfa020be6ac.png">

cansavvy replied:
Yes, so the 0's here are why we can't use both variables. 

I'm a bit confused on their experimental model (like I said in the original comment I haven't looked into their paper or descriptions much), but let's not go to far in the weeds and just use one experimental variable or the other (probably `refinebio_treatment` since it is already ready to go) And then probably just add a sentence telling the user what the variable is and why we are using it? We aren't doing DE here so in the end it doesn't matter. (But still have both labeled in the heatmap).

cansavvy replied:
So basically return to how you had it 😄  (but I think this has still been a good idea to look into what our dataset has). :+1:

jaclyn-taroni replied:
Is there a reason we shouldn't explicitly blind to experimental design/use no design with `~ 1` here? (Mentioned: https://github.com/AlexsLemonade/refinebio-examples/issues/127#issuecomment-658926700)

_Side note:_ If we're in a spot where we need to include direct messages during review of a pull request, it suggests to me that the issue https://github.com/AlexsLemonade/refinebio-examples/issues/127#issuecomment-669347715 could have included more context for why the course of action was:

> Said draft PR will include a more simplified code for the clustering example notebook, that would start with the heatmap, then deal with the metadata and annotation bars one by one.

cansavvy commented:
@cbethell , I couldn't replicate your pandoc error you mentioned (I ran it on the docker container from what's being developed on #159). 

Once #159  is merged, can you run the docker container and try running it on your end and see if you still get the same error?

jashapiro commented on Snakefile:
```diff
@@ -1,6 +1,7 @@
     input: 
-        "01-getting-started/getting-started.html" 
+        "01-getting-started/getting-started.html",
+	"03-rnaseq/clustering/clustering_heatmap_example.html" 
```
Snakemake, being based on python, cares a lot about spacing.
```suggestion
	  "03-rnaseq/clustering/clustering_heatmap_example.html" 
```

cbethell replied:
Ahh of course!

cansavvy commented:
With #146 being merged, you can use the template to get this analysis finalized. 
And with #158 merged, you can start up a container and render everything with snakemake!

Let me know how the instructions work for you.

cbethell commented:
> @cbethell with #146 being merged, you can use the template to get this analysis finalized.
> And with #158 merged, you can start up a container and render everything with snakemake!
> 
> Let me know how the instructions work for you.

The instructions worked great! @cansavvy (It may be useful however to add a note that the `screenshots` folder that is in the `template` directory will need to be copied over to the new module as well in order for snakemake to run successfully)

On another note, I believe I was able to format everything accordingly and render with snakemake! Please feel free to let me know if you think I missed any important steps, misinterpreted any of your suggestions, or should make any changes!

cansavvy commented:
Great! I will take a look! 

> (It may be useful however to add a note that the `screenshots` folder that is in the `template` directory will need to be copied over to the new module as well in order for snakemake to run successfully)

Did you see the CONTRIBUTING.md bit about that: https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#adding-a-new-analysis

If yes, is there a way I can make it more clear?

cbethell commented:
> Great! I will take a look!
> 
> > (It may be useful however to add a note that the `screenshots` folder that is in the `template` directory will need to be copied over to the new module as well in order for snakemake to run successfully)
> 
> Did you see the CONTRIBUTING.md bit about that: https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#adding-a-new-analysis
> 
> If yes, is there a way I can make it more clear?

Ah right! I saw it yesterday and had to read the sentence a few times because of the wording but in the end could not think of a better suggestion, of course how could I forget!

Perhaps we can change `Copy, paste and rename the template/template_example.Rmd file to start a new analysis and the template/screenshots folder` to `Copy, paste and rename the template/template_example.Rmd file and the template/screenshots folder to start a new analysis`

However, this was definitely an overlook on my behalf so it may not be as necessary to change right now as we may think as I am the only one who has tested the instructions thus far and can very well be the only one to _overlook/misread_ that sentence.

cansavvy commented:
> Perhaps we can change `Copy, paste and rename the template/template_example.Rmd file to start a new analysis and the template/screenshots folder` to `Copy, paste and rename the template/template_example.Rmd file and the template/screenshots folder to start a new analysis`
> 
> However, this was definitely an overlook on my behalf so it may not be as necessary to change right now as we may think as I am the only one who has tested the instructions thus far and can very well be the only one to _overlook/misread_ that sentence.

No worries, its definitely wordy. I will fix it! Thanks for the suggestions.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+## About the dataset we are using for this example
+
+For this example analysis, we will use this [acute myeloid leukemia sample dataset](https://alexslemonade.github.io/refinebio-examples/clustering/data/SRP070849.tsv).
```
Oh this was supposed to be  a link to the dataset on refinebio. But I see in my edits the prompt doesn't say that. That's my fault. Will fix in the template. Can you make this the refinebio dataset link in the meantime?

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+If you'd like to adapt an example analysis to use a different dataset from [refine.bio](https://www.refine.bio/), we recommend placing the files in the `data/` directory you created and changing the filenames and paths in the notebook to match these files (we've put comments to signify where you would need to change the code).
+We suggest saving plots and results to `plots/` and `results/` directories, respectively, as these are automatically created by the notebook.
+From here you can customize this analysis example to fit your own scientific questions and preferences. 
+# Clustering Heatmap - RNA-seq
```
Can you keep the extra line above here? And maybe even bring back the line (***)

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+library(DESeq2)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
```
In the template we switched to `library(magrittr)`. Did you change this or did the template not have this update?

cbethell replied:
Because this notebook was open before the template, I adapted the notebook instead of copying the template and adapting that so this was a remnant of the old `clustering_heatmap_example.Rmd` notebook (before the template).

cansavvy replied:
Okey dokey. No worries. Figured something like that.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+## Create a DESeqDataset
+
+We will be using the `DESeq2` package for normalization, which requires us to format our data into a `DESeqDataSet` object.
+We turn the data.frame (or matrix) into a [`DESeqDataSet` object](TODO: Add link to header section) and specify which variable labels our experimental groups using the `design` argument.
```
```suggestion
We turn the data frame (or matrix) into a [`DESeqDataSet` object](TODO: Add link to header section) and specify which variable labels our experimental groups using the `design` argument.
```

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+We will be using the `DESeq2` package for normalization, which requires us to format our data into a `DESeqDataSet` object.
+We turn the data.frame (or matrix) into a [`DESeqDataSet` object](TODO: Add link to header section) and specify which variable labels our experimental groups using the `design` argument.
+In this chunk of code, we will specify the `refinebio-treatment` variable to label our experimental groups because it is included in the experimental design as noted in the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/28193779/).[@AML]
```
```suggestion
In this chunk of code, we will specify the `refinebio_treatment` variable to label our experimental groups because it is included in the experimental design as noted in the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/28193779/).[@AML]
```
What does this actually mean? Can we switch this to a short description of what the information `refinebio_treatment` represents? aka what was done to those samples?

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
````diff
@@ -0,0 +1,421 @@
+In this chunk of code, we will specify the `refinebio-treatment` variable to label our experimental groups because it is included in the experimental design as noted in the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/28193779/).[@AML]
+
+```{r}
+# The `DESeqDataSetFromMatrix` function needs the values to be converted to integers
````
```suggestion
# The `DESeqDataSetFromMatrix()` function needs the values to be converted to integers
```

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
````diff
@@ -0,0 +1,421 @@
+
+```{r}
+# Normalize the data in the `DESeqDataSet` object using the
+# `rlog` function from the `DESEq2` R package
````
```suggestion
# `rlog()` function from the `DESEq2` R package
```

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+We've created a heatmap but although our genes and samples are clustered, there
+is not much information that we can gather here because we did not provide the
+`pheatmap` function with annotation labels for our samples.
```
```suggestion
`pheatmap()` function with annotation labels for our samples.
```

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+### Save heatmap as a png
+
+You can easily switch this to save to a jpeg or tiff by changing the function 
```
Don't forget one sentence per line now.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+      startsWith(refinebio_title, "TET2") ~ "TET2",
+      startsWith(refinebio_title, "IDH2") ~ "IDH2",
+      startsWith(refinebio_title, "WT") ~ "WT",
+      TRUE ~ "unknown"
```
Can you add a comment explaining how this works? Aka if none of the above criteria are true, then we mark it as "unknown".

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+    mutation,
+    refinebio_treatment
+  ) %>%
+  # The `pheatmap` function requires that the rownames of our annotation object matches the column names of our `DESeaDataSet` object
```
```suggestion
  # The `pheatmap()` function requires that the row names of our annotation object matches the column names of our `DESeaDataSet` object
```

cansavvy replied:
I will add a tidbit in the style guide about when we use row names vs rownames()

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+### Create annotated heatmap
+
+You can create an annotated heatmap by providing our annotation object to the `annotation_col` argument of the `pheatmap` function.
```
```suggestion
You can create an annotated heatmap by providing our annotation object to the `annotation_col` argument of the `pheatmap()` function.
```

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+### Save annotated heatmap as a png
+
+You can easily switch this to save to a jpeg or tiff by changing the function and file name within the function to the respective file suffix.
```
```suggestion
You can easily switch this to save to a JPEG or TIFF by changing the function and file name within the function to the respective file suffix.
```

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
````diff
@@ -0,0 +1,421 @@
+You can easily switch this to save to a jpeg or tiff by changing the function and file name within the function to the respective file suffix.
+
+```{r}
+# Open a png file
````
```suggestion
# Open a PNG file
```

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+- [Make heatmaps in R with pheatmap](https://slowkow.com/notes/pheatmap-tutorial/) [@make-heatmaps]
+
+For more complex heatmaps, you may want to take a look at:
```
I like that you are including information to describe the resources but 1) Can you keep it in the same bullet point line? and 2) This description doesn't tell me much beyond what the title of `ComplexHeatmap` already told me 😄 .

cbethell replied:
@cansavvy I _believe_ I fixed the description. Let me know if the new description is suffice or if you think users may require more context in the notebook before clicking the link.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+Your example analysis folder should now look something like this (except with respective experiment accession id and analysis notebook name you are using): 
+
+<img src="screenshots/analysis-folder-structure.png" width=400>
```
I am changing these to URLs over on #166 . Sorry this didn't make it into the version you used. I'm trying to rethink parts of the template and how we should go about managing updates.

cbethell replied:
Updated this notebook to include the URLS as you have them in #166 and removed the `screenshots` folder from this module/PR 👍

cansavvy replied:
Ah, it's merged in master now, so you probably could have done a merge with master to get that updated. But this works too. :+1:

cbethell replied:
Oh I didn't think that PR updated this particular module but no worries!

cansavvy replied:
Oh DuH. You are right.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata [tab separated values ("tsv")](https://en.wikipedia.org/wiki/Tab-separated_values) file and a data tsv file.
```
```suggestion
Data downloaded from refine.bio include a metadata [tab separated values (TSV)](https://en.wikipedia.org/wiki/Tab-separated_values) file and a data TSV file.
```

cbethell replied:
This line was copied from the template so should probably be changed there as well @cansavvy (will add this to #165)

cansavvy replied:
Ah! Sorry about that. Will change now!

cansavvy replied:
Thanks for the heads up!

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata [tab separated values ("tsv")](https://en.wikipedia.org/wiki/Tab-separated_values) file and a data tsv file.
+This chunk of code will read in both tsv files and add them as data.frames to your environment.
```
```suggestion
This chunk of code will read in both TSV files and add them as data frames to your environment.
```

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
````diff
@@ -0,0 +1,421 @@
+This chunk of code will read in both tsv files and add them as data.frames to your environment.
+
+```{r}
+# Read in metadata tsv file
````
```suggestion
# Read in metadata TSV file
```

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+  "metadata_SRP070849.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data tsv file
```
```suggestion
# Read in data TSV file
```

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
````diff
@@ -0,0 +1,421 @@
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+# How are our samples clustering together?
````
Is this a remnant from before? I think you can delete it now, because this information has been covered above.

jaclyn-taroni commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+## Obtain the `.Rmd` file of this analysis
+
+To run this example yourself, [download the `.Rmd` for this analysis by clicking this link](https://alexslemonade.github.io/refinebio-examples/ <<RELATIVE PATH TO THIS ANALYSIS FROM REFINEBIO-EXAMPLES>> ). 
```
This needs to be altered for this notebook I think

cbethell replied:
Good catch, on it 👍

cbethell commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,421 @@
+
+## Obtain the `.Rmd` file of this analysis
+
+To run this example yourself, [download the `.Rmd` for this analysis by clicking this link](https://alexslemonade.github.io/refinebio-examples/ <<RELATIVE PATH TO THIS ANALYSIS FROM REFINEBIO-EXAMPLES>> ). 
```
I searched for `<<` and replaced all of them with the appropriate links at once, but had skipped this one the first time around so it ended getting lost in all of the changes. I believe searching for `<<` is a pretty efficient method but I will definitely try to think of a possible improvement.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,416 @@
+dds <- DESeqDataSetFromMatrix(
+  countData = df, # This is the data.frame with the counts values for all replicates in our dataset
+  colData = metadata, # This is the data.frame with the annotation data for the replicates in the counts data.frame
+  design = ~refinebio_treatment # Replace with an appropriate design variable for your analysis
```
Let's go with @jaclyn-taroni 's plan of using `~1` since we don't use the model anyway. And maybe put a comment about that means we aren't specifying a model.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,416 @@
+
+We will be using the `DESeq2` package for normalization, which requires us to format our data into a `DESeqDataSet` object.
+We turn the data frame (or matrix) into a [`DESeqDataSet` object](TODO: Add link to header section) and specify which variable labels our experimental groups using the `design` argument.
+From the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/28193779/), we know that the mice with `IDH2` mutant AML were treated with vehicle or AG-221 (the first small molecule in vivo inhibitor of IDH2 to enter clinical trials) and the mice with `TET2` mutant AML were treated with vehicle or 5-Azacytidine (Decitabine, hypomethylating agent).[@AML]
```
This is a great run down! But now that we aren't going to be using this info in the DESeq2 object, can you put this in the `Prepare metadata for annotation` section? I think that would be a good place for it.

cansavvy commented on 03-rnaseq/clustering/clustering_heatmap_example.Rmd:
```diff
@@ -0,0 +1,416 @@
+We will be using the `DESeq2` package for normalization, which requires us to format our data into a `DESeqDataSet` object.
+We turn the data frame (or matrix) into a [`DESeqDataSet` object](TODO: Add link to header section) and specify which variable labels our experimental groups using the `design` argument.
+From the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/28193779/), we know that the mice with `IDH2` mutant AML were treated with vehicle or AG-221 (the first small molecule in vivo inhibitor of IDH2 to enter clinical trials) and the mice with `TET2` mutant AML were treated with vehicle or 5-Azacytidine (Decitabine, hypomethylating agent).[@AML]
+In this chunk of code, we will specify the `refinebio_treatment` variable to label our experimental groups as it contains the data on the specific treatment given for each sample as a part of the experimental design briefly described above.
```
Then we can replace this sentence with something that maybe links out some relevant information about specifying models for DESeq2 and that we are using 1 because we aren't doing any DE test here.

cbethell replied:
Gotcha!

cbethell commented:
> This looks great! Just a few more quick changes!

@cansavvy, made those above changes that you suggested! 

I am wondering however if we should also link to the `DESeq2` docs to fill in the TODO comment at line 205 `[`DESeqDataSet` object](TODO: Add link to header section)`. What do you think?

cansavvy commented:
> If so, would we also want to link to docs to fill in the TODO at line 259:

Because this repo isn't "live" in the sense that we don't link to it on refine.bio I'm fine with leaving the TODOs on there for now and once we further determine the content for and write up the RNA-seq header section #125 we will be able to come back and know where those links should go.

cansavvy commented:
I think this is close to being ready for merging. However, I've made some minor edits to the template (#165) as I've been reviewing this and those changes should also probably be over here. So once that is approved, if @cbethell doesn't mind, I will copy those changes to this clustering notebook and then I think this will be ready from my end. (TODOs are noted on https://github.com/AlexsLemonade/refinebio-examples/issues/125#issuecomment-673095903 )

cbethell commented:
> Final render and Rmd looks great! But shoot, I just noticed the reference.bib adds you put in are not an alphabetical order. I think that's all and then this is ready for merge. 🚀

Fixed it in the [last commit](https://github.com/AlexsLemonade/refinebio-examples/pull/151/commits/bbd937b71e7c2f3489ea00b5e29b922903c66658)! I also changed the `complex-heatmaps` citation name (which points to the manual) to `make-complex-heatmaps` so that it would be better distinguishable from the `ComplexHeatmap` citation name which points to the docs.
